sdk/src/wallet/account/operations/balance.rs

```diff
@@ -275,17 +96,258 @@ impl Account {
+                                        }
+                                    }
+
+                                    // Add native tokens
+                                    if let Some(native_tokens) = output.native_tokens() {
+                                        context.total_native_tokens.add_native_tokens(native_tokens.clone())?;
+                                    }
+                                } else {
+                                    // only add outputs that can't be locked now and at any point in the future
+                                    context.balance.potentially_locked_outputs.insert(*output_id, true);
+                                }
+                            } else {
+                                // Don't add expired outputs that can't ever be unlocked by us
+                                if let Some(expiration) = output
+                                    .unlock_conditions()
+                                    .expect("output needs to have unlock conditions")
+                                    .expiration()
+                                {
+                                    // Not expired, could get unlockable when it's expired, so we insert it
+                                    if local_time < expiration.timestamp() {
+                                        context.balance.potentially_locked_outputs.insert(*output_id, false);
+                                    }
+                                } else {
+                                    context.balance.potentially_locked_outputs.insert(*output_id, false);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    // TODO address param or another method?
+    /// Get the AccountBalance
+    pub async fn balance(&self, address: Option<Bech32Address>) -> crate::wallet::Result<AccountBalance> {
+        log::debug!("[BALANCE] get balance");
+
+        let account_details = self.details().await;
+        let network_id = self.client().get_network_id().await?;
+        let rent_structure = self.client().get_rent_structure().await?;
+        let mut context = BalanceContext::new(&account_details, network_id, rent_structure);
+
+        #[cfg(feature = "participation")]
+        {
+            // TODO what should this do when we're asking for an address balance?
+            context.balance.base_coin.voting_power = self.get_voting_power().await?;
+        }
+
+        if let Some(address) = address {
+            if let Some(address_with_unspent_outputs) = account_details
+                .addresses_with_unspent_outputs
+                .iter()
+                .find(|a| a.address == address)
+            {
+                self.address_balance(address_with_unspent_outputs, &mut context).await?;
+            } else {
+                // TODO error or 0 balance ?
```

Comment:
I'd say error, can be confusing otherwise since an address can have balance and would show as 0 just because it doesn't belong to the account

Reply:
I agree